CleanCodeApp/Modules/Features/Lua/ResetPassword/LuaResetPasswordViewController.swift

```diff
@@ -23,150 +29,192 @@ class LuaResetPasswordViewController: UIViewController {
     open override var preferredStatusBarStyle: UIStatusBarStyle {
         return .lightContent
     }
-
+    
     @IBAction func closeButtonAction(_ sender: Any) {
         dismiss(animated: true)
     }
-
-    @IBAction func recoverPasswordButton(_ sender: Any) {
-        if recoveryEmail {
+    
+    @IBAction func onPasswordRecoveryButtonTapped(_ sender: Any) {
+        if hasRequestedRecovery {
             dismiss(animated: true)
             return
         }
-
-        if validateForm() {
-            self.view.endEditing(true)
-            if !ConnectivityManager.shared.isConnected {
-                Globals.showNoInternetCOnnection(controller: self)
+        let isEmailAndConnectivityValid = validateEmailFormAndConnectivity()
```

Comment:
eu nao curto muito a ideia de um funcao que tem AND
pq deixa a ideia que já faz mais de uma coisa, quebrando o SRP
eu sugeriria:
do {
try validatePasswordRecovery()
view.endEditing(true)
sendPasswordResetRequest(parameters: passwordResetParameters)
} catch {
//mostraria um alerta com mensagem que varia de acordo com o erro (internet, ou email invalido)}
|

eu passaria a linha :
let passwordResetParameters = getPasswordResetRequestParameters()
para dentro de sendPasswordResetRequest, pq ela ja esta bem pequena, e faz sentido os parâmetros estarem juntos da chamada (nesse caso)